end-to-end-untargeted-metabolomics.Rmd

```diff
@@ -1656,39 +1674,331 @@ plotting does not really bring a lot.
-have a CV < 31%.
+```{r}
+save(data, file = "data_afternorm.RData")
+save(res, file = "SumExp_afternorm.RData")
+```
 
 ### Conclusion
 
 No much difference but still the RLA plot showed that it allowed to centered the
 data around the median...  for bigger and more complex datatset...
 
-
 # Identification of interesting features
 
-- Pre-filtering.
-- Differential abundance analysis.
-    r Biocpkg("limma") or other packages
+number of features
+```{r}
+load("SumExp_afternorm.RData")
+load("data_afternorm.RData")
+#' Number of features before filtering
+nrow(res)
+
+#' keep unfitered object
+res_unfilt <- res
+```
+
+## Quality control
+
+After normalizing our data we can now pre-filter to clean the data before
+performing any statistical analysis. In general, pre-filtering of samples and features is performed to remove outliers.
+
+### Feature prefiltering
+
+Here we will remove the the features that have
+QC samples are usually the best to "clean" our dataset or systematic and
+random errors. In our dataset, because our QC pool samples are NOT pools of
+the samples but pool of other biological samples, we are more limited in their
+use for filtering. A in-depth review of the guidelines for data filtering of
+untargeted metabolomic studies can be found here:...
+
+We will however still base our pre-filtering on the QC sample of the base that
+they are similar bio-fluids of our study samples and therefore relatively
+similar metabolites are expected to be seen.
+Considering our QC samples, we will first use the previously calculated
+coefficient of variation and filter out all features with a coefficient above
+30% in these.
+
+We first restrict the data set to features for which a chromatographic peak was
+detected in at least 2/3 of samples of at least one of the study samples groups.
+This ensures the statistical tests performed later on the study samples being
+performed on *reliable* signal. Also, with this filter we remove features that
+were mostly detected in QC samples, but not the study samples. Such filter can
+be performed with `filterFeatures` and the `PercentMissingFilter`
+setting. Parameter `threshold` of this filter defines the maximal acceptable
+percentage of samples with missing value(s) in at least in one of the sample
+groups defined by parameter `f`. To consider detected chromatographic peaks per
+sample, we apply the filter on the `"raw"` assay of our result object, that
+contains an abundance value only for detected chromatographic peaks (prior
+gap-filling). By replacing the `"QC"` sample group for parameter `f` we consider
+only study samples. With `threshold = 40` we remove features for which no peak
+was identified in 2 out of the 3 samples per sample group.
 
+```{r}
+#' Restrict to features that have at least two detected peaks in at least
+#' one of the **study** sample groups.
+f <- res$phenotype
+f[f == "QC"] <- NA
+f <- as.factor(f)
```

Comment:
do we need that ? the internal code of the function does this already. Or is it better to teach people to have a correct factor input ?